gradle-jdks/src/test/groovy/com/palantir/gradle/jdks/GradleJdkToolchainsIntegrationTest.groovy

```diff
@@ -254,6 +256,141 @@ class GradleJdkToolchainsIntegrationTest extends GradleJdkIntegrationSpec {
 
         where:
         gradleVersionNumber << GRADLE_TEST_VERSIONS
+
+    }
+
+    def '#gradleVersionNumber: can bump java major version when baseline-java is applied'() {
+        gradleVersion = gradleVersionNumber
+        setupJdksHardcodedVersions('11')
+        applyBaselineJavaVersions()
+        applyApplicationPlugin()
+
+        // language=groovy
+        buildFile << """
+            javaVersions {
+                libraryTarget = '11'
+            }
+        """.stripIndent(true)
+
+        file('gradle.properties') << 'palantir.jdk.setup.enabled=true'
+        file('src/main/java/Main.java') << java17PreviewCode
+        runTasksSuccessfully('wrapper')
+
+        when:
+        runTasksSuccessfully('generateGradleJdkConfigs', '--includeVersion=11')
+
+        then: 'generates directories for jdk version == 11'
+        Files.list(projectDir.toPath().resolve("gradle/jdks"))
+                .map { it -> it.getFileName().toString() }
+                .collect(Collectors.toList())
+                .containsAll(List.of("11"))
```

Comment:
would you not want it to be a stricter assertion i.e. it's *only* `11` or are there other jdks inserted? I can't fully tell yet